test/parallel/test-cluster-worker-wait-server-close.js

```diff
@@ -22,10 +22,12 @@ if (cluster.isWorker) {
   worker.once('listening', function() {
     net.createConnection(common.PORT, common.localhostIPv4, function() {
       var socket = this;
-      worker.disconnect();
       setTimeout(function() {
-        socket.write('.');
-        connectionDone = true;
+        worker.disconnect();
```

Comment:
@trevnorris the disconnect callback will never be called, not while there is an open connection to the worker. the test calls disconnect, and waits one second to confirm that disconnect did NOT in fact, cause disconnection... only when the socket is closed, triggered by the write('.'), does the disconnect proceed.